### PR TITLE
Add password e2e test

### LINKED ISSUE #4326: Create a random password after TiDB installation
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, TiDB will create a root user with no password by default. This is a potential security issue.
So we have to create a random password after TiDB is installed.
We can add a field in the spec to control whether to create a random password after installation, if yes, we can save the password in a secret and request the user to access the TiDB with that password and then change the password.
TiDB will force the user to change the password after the first login in future releases.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Add password e2e test.
Fix #4326 
### What is changed and how does it work?
Add password e2e test.
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -3216,6 +3219,55 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 		err = cli.PingcapV1alpha1().TidbMonitors(tm.Namespace).Delete(context.TODO(), tm.Name, metav1.DeleteOptions{})
 		framework.ExpectNoError(err, "delete tidbmonitor failed")
 	})
+
+	ginkgo.It("deploy tidb cluster with random password", func() {
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:3223]: Please also add a TLS case.

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -3216,6 +3219,55 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
+
+		tc, err := cli.PingcapV1alpha1().TidbClusters(tc.Namespace).Create(context.TODO(), tc, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Expected create tidbcluster")
+		err = oa.WaitForTidbClusterReady(tc, 30*time.Minute, 5*time.Second)
+		framework.ExpectNoError(err, "Expected get tidbcluster")
+
+		err = wait.Poll(10*time.Second, 3*time.Minute, func() (done bool, err error) {
+			randomPasswordTc, err := cli.PingcapV1alpha1().TidbClusters(tc.Namespace).Get(context.TODO(), tcName, metav1.GetOptions{})
+			if err != nil {
+				klog.Errorf("Failed to get TidbCluster[%s:%s],error:%v", ns, tcName, err)
```
Comment (csuzhangxc) [inline tests/e2e/tidbcluster/tidbcluster.go:3240]:
```suggestion
				klog.Errorf("Failed to get TidbCluster[%s:%s], error:%v", ns, tcName, err)
```

File: tests/actions.go [hunk 3]
```diff
@@ -3697,6 +3698,47 @@ func (oa *OperatorActions) WaitForTidbClusterReady(tc *v1alpha1.TidbCluster, tim
 	return err
 }
 
+func (oa *OperatorActions) WaitForConnectTidbCluster(tc *v1alpha1.TidbCluster, timeout, pollInterval time.Duration) error {
```
Comment (csuzhangxc) [inline tests/actions.go:3701]: this method only works for TC with `PasswordInitialized` right? Can we give it a meaningful name?

**Review by csuzhangxc — APPROVED** [review]

File: tests/actions.go (hunk 3 above)
Comment (mikechengwei) [reply tests/actions.go]: ok

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@mikechengwei Please fix the CI.

**Pull request merged**